File: api/js/src/validator/diagnostics.ts
import type {
  Diagnostic,
  DiagnosticLocation,
  DiagnosticSeverity,
} from "./types.js";

export function createDiagnostic(
  ruleId: string,
  message: string,
  severity: DiagnosticSeverity,
  location?: DiagnosticLocation,
  hint?: string,
): Diagnostic {
  const diagnostic: Diagnostic = { ruleId, message, severity };
  if (location !== undefined) {
    diagnostic.location = location;
  }
  if (hint !== undefined) {
    diagnostic.hint = hint;
  }
  return diagnostic;
}

File: api/python/telegram_inline_keyboard_builder/validator/diagnostics.py
from __future__ import annotations

from .types import Diagnostic, DiagnosticLocation, DiagnosticSeverity


def create_diagnostic(
    rule_id: str,
    message: str,
    severity: DiagnosticSeverity,
    location: DiagnosticLocation | None = None,
    hint: str | None = None,
) -> Diagnostic:
    diagnostic: Diagnostic = {
        "rule_id": rule_id,
        "message": message,
        "severity": severity,
    }
    if location is not None:
        diagnostic["location"] = location
    if hint is not None:
        diagnostic["hint"] = hint
    return diagnostic
